repository.py

```diff
@@ -130,14 +130,16 @@ def _parse_repodata(self, accessor):
         repomdfp.close()
 
         primaryfp = accessor.openAddress(primary_location)
-        # Open compressed xml using cpiofile._Stream which is an adapter between CpioFile and a stream-like object.
-        # Useful when specifying the URL for HTTP or FTP repository - A simple GzipFile object will not work in this situation.
-        primary_xml = cpiofile._Stream("", "r", "gz", primaryfp, 20*512)
-        primary_dom = parse(primary_xml)
-        package_names = primary_dom.getElementsByTagName("location")
-        package_sizes = primary_dom.getElementsByTagName("size")
-        package_checksums = primary_dom.getElementsByTagName("checksum")
-        primary_xml.close()
+
+        # HTTP/FTP accessors don't implement tell(), so read the data into a buffer
+        with BytesIO(primaryfp.read()) as fileobj:
+            with gzip.GzipFile("", "r", fileobj=fileobj) as primary_xml:
+                dom = parse(primary_xml)
+                package_names = dom.getElementsByTagName("location")
+                package_sizes = dom.getElementsByTagName("size")
+                package_checksums = dom.getElementsByTagName("checksum")
+                dom.unlink()
```

Comment:
The problem with `dom.unlink()` is that it releases too much memory 😆 
In particular the checksums are empty so installation will fail.

Reply:
Doh! 😆

I forgot to check that the return values of getElementsByTagName() are still part of the DOM and not copied elements or strings!

`dom.unlink()` removed and updated the commit message accordingly!